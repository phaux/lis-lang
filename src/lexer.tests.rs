use super::*;

#[test]
fn prog_simple() {
    let input = "let x = 5;";
    let tokens: Vec<_> = Lexer::new(input).collect();
    insta::assert_debug_snapshot!(tokens);
}

#[test]
fn assign_str_escaped() {
    let input = r#" s = "hello\nworld\twith\\quotes\"" "#;
    let tokens: Vec<_> = Lexer::new(input).collect();
    insta::assert_debug_snapshot!(tokens);
}

#[test]
fn assign_str_unclosed() {
    let input = r#" s = "hello "#;
    let tokens: Vec<_> = Lexer::new(input).collect();
    insta::assert_debug_snapshot!(tokens);
}

#[test]
fn assign_num() {
    let input = " x = 5.5 ";
    let tokens: Vec<_> = Lexer::new(input).collect();
    insta::assert_debug_snapshot!(tokens);
}

#[test]
fn assign_num_invalid() {
    let input = " x = 5.5.5 ";
    let tokens: Vec<_> = Lexer::new(input).collect();
    insta::assert_debug_snapshot!(tokens);
}

#[test]
fn comment_line() {
    let input = "let x = 1; // this is a comment\nprint x";
    let tokens: Vec<_> = Lexer::new(input).collect();
    insta::assert_debug_snapshot!(tokens);
}

#[test]
fn comment_block() {
    let input = "let x = /* block comment */ 1;";
    let tokens: Vec<_> = Lexer::new(input).collect();
    insta::assert_debug_snapshot!(tokens);
}
