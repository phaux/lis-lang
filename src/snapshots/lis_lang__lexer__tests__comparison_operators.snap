---
source: src/lexer.tests.rs
expression: tokens
---
[
    Token {
        sigil: Ident {
            name: "a",
        },
        offset: 0..1,
        pos: Pos {
            line: 0,
            col: 0,
        }..Pos {
            line: 0,
            col: 1,
        },
    },
    Token {
        sigil: Less,
        offset: 2..3,
        pos: Pos {
            line: 0,
            col: 2,
        }..Pos {
            line: 0,
            col: 3,
        },
    },
    Token {
        sigil: Ident {
            name: "b",
        },
        offset: 4..5,
        pos: Pos {
            line: 0,
            col: 4,
        }..Pos {
            line: 0,
            col: 5,
        },
    },
    Token {
        sigil: LessEq,
        offset: 6..8,
        pos: Pos {
            line: 0,
            col: 6,
        }..Pos {
            line: 0,
            col: 8,
        },
    },
    Token {
        sigil: Ident {
            name: "c",
        },
        offset: 9..10,
        pos: Pos {
            line: 0,
            col: 9,
        }..Pos {
            line: 0,
            col: 10,
        },
    },
    Token {
        sigil: Greater,
        offset: 11..12,
        pos: Pos {
            line: 0,
            col: 11,
        }..Pos {
            line: 0,
            col: 12,
        },
    },
    Token {
        sigil: Ident {
            name: "d",
        },
        offset: 13..14,
        pos: Pos {
            line: 0,
            col: 13,
        }..Pos {
            line: 0,
            col: 14,
        },
    },
    Token {
        sigil: GreaterEq,
        offset: 15..17,
        pos: Pos {
            line: 0,
            col: 15,
        }..Pos {
            line: 0,
            col: 17,
        },
    },
    Token {
        sigil: Ident {
            name: "e",
        },
        offset: 18..19,
        pos: Pos {
            line: 0,
            col: 18,
        }..Pos {
            line: 0,
            col: 19,
        },
    },
]
